.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -2,62 +2,55 @@
-  done and why you've done it? Can other users understand how to use your changes?
-If not but opening the pull request will facilitate development, make it a "draft" pull request
-
-This form is written in GitHub's Markdown format. For a reference on this type
-of syntax, see GitHub's documentation:
-https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax
-
-This template contains guidance for your submission within the < ! - -, - - > blocks.
-These are comments in HTML syntax and will not appear in the submission.
-Be sure to use the "Preview" feature on GitHub to ensure your submission is formatted as intended.
-
-When including code snippets, please paste the text itself and wrap the code block with
-ticks (see the other character on the tilde ~ key in a US keyboard) to format it as code.
-For example, Python code should be wrapped in ticks like this:
-```python
-def a_func():
-    return 1
-
-a = 1
-b = a_func()
-print(a + b)
-```
-This is preferred over screen shots since it is searchable and others can copy/paste
-the text to run it.
--->
+1. Use GH flavored markdown when writing your description:
+   https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax
 
+2. If all checkboxes in the PR Checklist cannot be merged, this PR should be marked as a draft.
 
-# Add meaningful title here
-<!--
-Be sure to title your pull request so that readers can scan through the list of PR's and understand
-what this one involves. It should be a few well selected words to get the point across. If you have
-a hard time choosing a brief title, consider splitting the pull request into multiple pull requests.
-Keep in mind that the title will be automatically included in the release notes.
+3. Do not delete anything from this template. If a section does not apply to you, simply write
+   "N/A" in the description.
+
+4. Code snippets to highlight new, modified, or problematic functionality are highly encouraged,
+   though not required. Be sure to use proper code higlighting as demonstrated below.
+
+   ```python
+    def a_func():
+        return 1
+
+    a = 1
+    b = a_func()
+    print(a + b)
+    ```
 -->
-Describe your feature here.
+
+# Add meaningful title here
+<!--The title should clearly define your contribution succinctly.-->
+
+Describe your feature here. Please include any code snippets or examples in this section.
+
+## PR Checklist
+<!--Tick these boxes if they are complete.-->
+
+- [ ] Documentation
+  - [ ] Docstrings are up-to-date
+  - [ ] Related `docs/` files are up-to-date, or added when necessary
+  - [ ] Documentation has been rebuilt successfully
+  - [ ] Examples have been updated
+- [ ] Tests pass (If not, and this is expected please elaborate in the tests section)
+- [ ] PR description thoroughly describes the new feature, bug fix, etc.
```

Comment:
Love this, I chatted with Raf and he said something like this was quite helpful for floris dev